drivers/pinctrl/pfc_rcar.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023 IoT.bzh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_PINCTRL_PFC_RCAR_H_
+#define ZEPHYR_DRIVERS_PINCTRL_PFC_RCAR_H_
+
+const struct pfc_bias_reg *pfc_rcar_get_bias_regs(void);
+const struct pfc_drive_reg *pfc_rcar_get_drive_regs(void);
+
+/**
+ * @brief set the register index for a given pin
+ *
+ * @param the pin
+ * @param pointer for the resulting register index
+ * @return 0 if the register index is found, negative
+ * errno otherwise.
+ */
+int pfc_rcar_get_reg_index(uint8_t pin, uint8_t *reg_index);
```

Comment:
this header is not self-contained

Reply:
Sorry, I was not aware of this rule

So, I moved the "pfc_rcar.h" include to the top of pfc_rcar.c includes as proposed in #41543 , it also allows me to test self-containment of this new header.

For it to be fully self-contained, I think I need to include pinctrl.soc that is defining both `pfc_bias_reg` and `pfc_drive_reg`  struct.
I'm asking this because, including stdint is enough for it to build (it seems like gcc is not checking struct def at build) but it seems wrong to me.

Researches led me to this paper[1] explaining that all "incomplete" types would be managed the same at build.

What would be Zephyr guideline in this case ?

[1] : https://beej.us/guide/bgc/html/split/incomplete-types.html